HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/main/README.md
Retrying in 1s [Retry 1/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/main/README.md
Retrying in 2s [Retry 2/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/main/README.md
Retrying in 4s [Retry 3/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/main/README.md
Retrying in 8s [Retry 4/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/main/README.md
Retrying in 8s [Retry 5/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/main/README.md
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/394113073258ead631f617d2e13bb377c0715c4b/pacs.py
Retrying in 1s [Retry 1/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/394113073258ead631f617d2e13bb377c0715c4b/pacs.py
Retrying in 2s [Retry 2/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/394113073258ead631f617d2e13bb377c0715c4b/pacs.py
Retrying in 4s [Retry 3/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/394113073258ead631f617d2e13bb377c0715c4b/pacs.py
Retrying in 8s [Retry 4/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/394113073258ead631f617d2e13bb377c0715c4b/pacs.py
Retrying in 8s [Retry 5/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/394113073258ead631f617d2e13bb377c0715c4b/pacs.py
Using the latest cached version of the dataset since flwrlabs/pacs couldn't be found on the Hugging Face Hub
Found the latest cached dataset configuration 'default' at /home/apiasecz/.cache/huggingface/datasets/flwrlabs___pacs/default/0.0.0/394113073258ead631f617d2e13bb377c0715c4b (last modified on Wed Jan 29 16:59:35 2025).
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/main/README.md
Retrying in 1s [Retry 1/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/main/README.md
Retrying in 2s [Retry 2/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/main/README.md
Retrying in 4s [Retry 3/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/main/README.md
Retrying in 8s [Retry 4/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/main/README.md
Retrying in 8s [Retry 5/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/main/README.md
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/394113073258ead631f617d2e13bb377c0715c4b/pacs.py
Retrying in 1s [Retry 1/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/394113073258ead631f617d2e13bb377c0715c4b/pacs.py
Retrying in 2s [Retry 2/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/394113073258ead631f617d2e13bb377c0715c4b/pacs.py
Retrying in 4s [Retry 3/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/394113073258ead631f617d2e13bb377c0715c4b/pacs.py
Retrying in 8s [Retry 4/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/394113073258ead631f617d2e13bb377c0715c4b/pacs.py
Retrying in 8s [Retry 5/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/394113073258ead631f617d2e13bb377c0715c4b/pacs.py
Using the latest cached version of the dataset since flwrlabs/pacs couldn't be found on the Hugging Face Hub
Found the latest cached dataset configuration 'default' at /home/apiasecz/.cache/huggingface/datasets/flwrlabs___pacs/default/0.0.0/394113073258ead631f617d2e13bb377c0715c4b (last modified on Wed Jan 29 16:59:35 2025).
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/main/README.md
Retrying in 1s [Retry 1/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/main/README.md
Retrying in 2s [Retry 2/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/main/README.md
Retrying in 4s [Retry 3/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/main/README.md
Retrying in 8s [Retry 4/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/main/README.md
Retrying in 8s [Retry 5/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/main/README.md
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/394113073258ead631f617d2e13bb377c0715c4b/pacs.py
Retrying in 1s [Retry 1/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/394113073258ead631f617d2e13bb377c0715c4b/pacs.py
Retrying in 2s [Retry 2/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/394113073258ead631f617d2e13bb377c0715c4b/pacs.py
Retrying in 4s [Retry 3/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/394113073258ead631f617d2e13bb377c0715c4b/pacs.py
Retrying in 8s [Retry 4/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/394113073258ead631f617d2e13bb377c0715c4b/pacs.py
Retrying in 8s [Retry 5/5].
HTTP Error 429 thrown while requesting HEAD https://huggingface.co/datasets/flwrlabs/pacs/resolve/394113073258ead631f617d2e13bb377c0715c4b/pacs.py
Using the latest cached version of the dataset since flwrlabs/pacs couldn't be found on the Hugging Face Hub
Found the latest cached dataset configuration 'default' at /home/apiasecz/.cache/huggingface/datasets/flwrlabs___pacs/default/0.0.0/394113073258ead631f617d2e13bb377c0715c4b (last modified on Wed Jan 29 16:59:35 2025).
Device: cuda
CUDA Version: 12.1
cuDNN Version: 8902
PyTorch Version: 2.3.0

Model used: PACSCNN
Initial dataset creation
Initial dataset creation
Initial loss: 26.615753173828125
<__main__.Policy object at 0x14606f946210>
26.615753173828125
Loss historical diff: 0.0; Loss Difference: 0.0, Decision: 0
Round 1 took 4.748927116394043s, results:
{'t': 0, 'current_accuracy': 0.109375, 'current_loss': 25.071142196655273, 'train_loss': None, 'decision': 0, 'drift_rate': 0.0, 'target_domains': ['art_painting']}
Loss historical diff: 0.0; Loss Difference: 0.0, Decision: 0
Round 2 took 4.496119499206543s, results:
{'t': 1, 'current_accuracy': 0.109375, 'current_loss': 25.07113742828369, 'train_loss': None, 'decision': 0, 'drift_rate': 0.0, 'target_domains': ['art_painting']}
Loss historical diff: 2.86102294921875e-06; Loss Difference: 0.0, Decision: 0
Round 3 took 4.511982202529907s, results:
{'t': 2, 'current_accuracy': 0.109375, 'current_loss': 25.07114028930664, 'train_loss': None, 'decision': 0, 'drift_rate': 0.0, 'target_domains': ['art_painting']}
Loss historical diff: 0.0; Loss Difference: 0.0, Decision: 0
Round 4 took 4.163874626159668s, results:
{'t': 3, 'current_accuracy': 0.109375, 'current_loss': 25.07113742828369, 'train_loss': None, 'decision': 0, 'drift_rate': 0.0, 'target_domains': ['art_painting']}
Loss historical diff: 0.0; Loss Difference: 0.0, Decision: 0
Round 5 took 4.262338638305664s, results:
{'t': 4, 'current_accuracy': 0.109375, 'current_loss': 25.07113742828369, 'train_loss': None, 'decision': 0, 'drift_rate': 0.0, 'target_domains': ['art_painting']}
Loss historical diff: 7.62939453125e-06; Loss Difference: 0.0, Decision: 0
Round 6 took 4.226820230484009s, results:
{'t': 5, 'current_accuracy': 0.109375, 'current_loss': 25.071145057678223, 'train_loss': None, 'decision': 0, 'drift_rate': 0.0, 'target_domains': ['art_painting']}
Loss historical diff: 9.5367431640625e-07; Loss Difference: 0.0, Decision: 0
Round 7 took 4.346370220184326s, results:
{'t': 6, 'current_accuracy': 0.109375, 'current_loss': 25.071138381958008, 'train_loss': None, 'decision': 0, 'drift_rate': 0.0, 'target_domains': ['art_painting']}
Loss historical diff: 1.9073486328125e-06; Loss Difference: 0.0, Decision: 0
Round 8 took 4.132792234420776s, results:
{'t': 7, 'current_accuracy': 0.109375, 'current_loss': 25.071139335632324, 'train_loss': None, 'decision': 0, 'drift_rate': 0.0, 'target_domains': ['art_painting']}
Loss historical diff: 5.7220458984375e-06; Loss Difference: 0.0, Decision: 0
Round 9 took 4.505964517593384s, results:
{'t': 8, 'current_accuracy': 0.109375, 'current_loss': 25.07114315032959, 'train_loss': None, 'decision': 0, 'drift_rate': 0.0, 'target_domains': ['art_painting']}
Loss historical diff: 0.0; Loss Difference: 0.0, Decision: 0
Round 10 took 4.875789403915405s, results:
{'t': 9, 'current_accuracy': 0.109375, 'current_loss': 25.07113742828369, 'train_loss': None, 'decision': 0, 'drift_rate': 0.0, 'target_domains': ['art_painting']}
Loss historical diff: 9.5367431640625e-07; Loss Difference: 0.0, Decision: 0
Round 11 took 4.306650638580322s, results:
{'t': 10, 'current_accuracy': 0.109375, 'current_loss': 25.071138381958008, 'train_loss': None, 'decision': 0, 'drift_rate': 0.0, 'target_domains': ['art_painting']}
Loss historical diff: 9.5367431640625e-07; Loss Difference: 0.0, Decision: 0
Round 12 took 4.294248342514038s, results:
{'t': 11, 'current_accuracy': 0.109375, 'current_loss': 25.071138381958008, 'train_loss': None, 'decision': 0, 'drift_rate': 0.0, 'target_domains': ['art_painting']}
Loss historical diff: 0.0; Loss Difference: 0.0, Decision: 0
Round 13 took 4.200868368148804s, results:
{'t': 12, 'current_accuracy': 0.109375, 'current_loss': 25.07113742828369, 'train_loss': None, 'decision': 0, 'drift_rate': 0.0, 'target_domains': ['art_painting']}
Loss historical diff: 1.9073486328125e-06; Loss Difference: 0.0, Decision: 0
Round 14 took 4.526039361953735s, results:
{'t': 13, 'current_accuracy': 0.109375, 'current_loss': 25.071139335632324, 'train_loss': None, 'decision': 0, 'drift_rate': 0.0, 'target_domains': ['art_painting']}
Loss historical diff: 0.0; Loss Difference: 0.0, Decision: 0
Round 15 took 4.515604257583618s, results:
{'t': 14, 'current_accuracy': 0.109375, 'current_loss': 25.071136474609375, 'train_loss': None, 'decision': 0, 'drift_rate': 0.0, 'target_domains': ['art_painting']}
Loss historical diff: 5.7220458984375e-06; Loss Difference: 0.0, Decision: 0
Round 16 took 4.7392048835754395s, results:
{'t': 15, 'current_accuracy': 0.109375, 'current_loss': 25.071142196655273, 'train_loss': None, 'decision': 0, 'drift_rate': 0.0, 'target_domains': ['art_painting']}
Loss historical diff: 0.0; Loss Difference: 0.0, Decision: 0
Round 17 took 4.57463264465332s, results:
{'t': 16, 'current_accuracy': 0.109375, 'current_loss': 25.071136474609375, 'train_loss': None, 'decision': 0, 'drift_rate': 0.0, 'target_domains': ['art_painting']}
Loss historical diff: 1.9073486328125e-06; Loss Difference: 0.0, Decision: 0
Round 18 took 4.140836954116821s, results:
{'t': 17, 'current_accuracy': 0.109375, 'current_loss': 25.071138381958008, 'train_loss': None, 'decision': 0, 'drift_rate': 0.0, 'target_domains': ['art_painting']}
Loss historical diff: 3.814697265625e-06; Loss Difference: 0.0, Decision: 0
Round 19 took 3.7975118160247803s, results:
{'t': 18, 'current_accuracy': 0.109375, 'current_loss': 25.07114028930664, 'train_loss': None, 'decision': 0, 'drift_rate': 0.0, 'target_domains': ['art_painting']}
Loss historical diff: 3.814697265625e-06; Loss Difference: 0.0, Decision: 0
Round 20 took 4.294842958450317s, results:
{'t': 19, 'current_accuracy': 0.109375, 'current_loss': 25.07114028930664, 'train_loss': None, 'decision': 0, 'drift_rate': 0.0, 'target_domains': ['art_painting']}
Loss historical diff: 3.113234519958496; Loss Difference: 0.0, Decision: 1
/home/apiasecz/.conda/envs/cog_fl_llm_env/lib/python3.12/site-packages/torch/nn/modules/conv.py:456: UserWarning: Plan failed with an OutOfMemoryError: CUDA out of memory. Tried to allocate 1.00 GiB. GPU  (Triggered internally at /opt/conda/conda-bld/pytorch_1712608847532/work/aten/src/ATen/native/cudnn/Conv_v8.cpp:924.)
  return F.conv2d(input, weight, bias, self.stride,
/home/apiasecz/.conda/envs/cog_fl_llm_env/lib/python3.12/site-packages/torch/nn/modules/conv.py:456: UserWarning: Plan failed with an OutOfMemoryError: CUDA out of memory. Tried to allocate 514.00 MiB. GPU  (Triggered internally at /opt/conda/conda-bld/pytorch_1712608847532/work/aten/src/ATen/native/cudnn/Conv_v8.cpp:924.)
  return F.conv2d(input, weight, bias, self.stride,
/home/apiasecz/.conda/envs/cog_fl_llm_env/lib/python3.12/site-packages/torch/nn/modules/conv.py:456: UserWarning: Plan failed with an OutOfMemoryError: CUDA out of memory. Tried to allocate 2.25 GiB. GPU  (Triggered internally at /opt/conda/conda-bld/pytorch_1712608847532/work/aten/src/ATen/native/cudnn/Conv_v8.cpp:924.)
  return F.conv2d(input, weight, bias, self.stride,
Traceback (most recent call last):
  File "/home/apiasecz/programming/concept_drift_optimal_adaptation_FL/src/execute/evaluate_policy.py", line 753, in <module>
    main()
  File "/home/apiasecz/programming/concept_drift_optimal_adaptation_FL/src/execute/evaluate_policy.py", line 733, in main
    evaluate_policy_under_drift(
  File "/home/apiasecz/programming/concept_drift_optimal_adaptation_FL/src/execute/evaluate_policy.py", line 554, in evaluate_policy_under_drift
    update_loss = agent_train.update_steps(
                  ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/apiasecz/programming/Federated_Learning_Base_Toolkit_torch/fl_toolkit/federated/client.py", line 202, in update_steps
    return self.model.update_steps(self.data_loader, optimizer, loss_fn, num_updates, verbose)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/apiasecz/programming/Federated_Learning_Base_Toolkit_torch/fl_toolkit/models/base_model.py", line 93, in update_steps
    outputs = self.model(inputs)
              ^^^^^^^^^^^^^^^^^^
  File "/home/apiasecz/.conda/envs/cog_fl_llm_env/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1532, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/apiasecz/.conda/envs/cog_fl_llm_env/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1541, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/apiasecz/programming/concept_drift_optimal_adaptation_FL/src/execute/evaluate_policy.py", line 47, in forward
    x = self.features(x)
        ^^^^^^^^^^^^^^^^
  File "/home/apiasecz/.conda/envs/cog_fl_llm_env/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1532, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/apiasecz/.conda/envs/cog_fl_llm_env/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1541, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/apiasecz/.conda/envs/cog_fl_llm_env/lib/python3.12/site-packages/torch/nn/modules/container.py", line 217, in forward
    input = module(input)
            ^^^^^^^^^^^^^
  File "/home/apiasecz/.conda/envs/cog_fl_llm_env/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1532, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/apiasecz/.conda/envs/cog_fl_llm_env/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1541, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/apiasecz/programming/concept_drift_optimal_adaptation_FL/src/execute/evaluate_policy.py", line 70, in forward
    out = self.bn1(out)
          ^^^^^^^^^^^^^
  File "/home/apiasecz/.conda/envs/cog_fl_llm_env/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1532, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/apiasecz/.conda/envs/cog_fl_llm_env/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1541, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/apiasecz/.conda/envs/cog_fl_llm_env/lib/python3.12/site-packages/torch/nn/modules/batchnorm.py", line 175, in forward
    return F.batch_norm(
           ^^^^^^^^^^^^^
  File "/home/apiasecz/.conda/envs/cog_fl_llm_env/lib/python3.12/site-packages/torch/nn/functional.py", line 2509, in batch_norm
    return torch.batch_norm(
           ^^^^^^^^^^^^^^^^^
torch.cuda.OutOfMemoryError: CUDA out of memory. Tried to allocate 512.00 MiB. GPU 
